src/js/dropdown-menu/dropdown-menu-view.js

```diff
@@ -130,9 +136,9 @@ define([
             },
 
             escKeypressHandler: function(event) {
-                var keyCode = event.keyCode;
+                var key = event.keyCode;
 
-                if (keyCode === constants.keyCodes.esc) {
+                if (_.contains(this.keyClose, key)) {
                     // When the ESC key is pressed, close all menus
                     this.closeDropdownMenus(true);
```

Comment:
Yes, this currently closes all dropdowns but I see your point about scope. I will update to only close the current view's menu.